fastparquet/dataframe.py

```diff
@@ -181,6 +186,9 @@ def set_cats(values, i=i, col=col, **kwargs):
             new_block = block.make_block_same_class(
                 type(block.values)(values, dtype=block.values.dtype)
             )
+        elif hasattr(block.values.dtype,'na_value'):
+            values = pd.array([None]*size, dtype=block.values.dtype)
```

Comment:
Possibly safer to use `[dtype.na_value]`? That seems a bit safer than assuming that the dtype will coerce None to the NA value.

Reply:
As suggested, I'm using block.values.dtype.na_value